api/components/docs/forms-and-redirects.md

```diff
@@ -0,0 +1,109 @@
+## Form Submission
+
+### `text/html` forms
+In XInput, `text/html` forms are used to capture additional information from user. 
+
+Buyer app submits the form to the action url provided in the form html. Seller app will do basic sanity checks like format of the inputs, required fields etc and respond with `status` as `SUBMITTED` and a `submission_id` which is used by the buyer app in the next action call. 
+If the sanity checks fail, seller app will respond with `NACK` with an appropriate error code. Buyer app can retry with updated data.
```

Comment:
If form submission is unsuccessful, we are sending `NACK` currently which has only one error attribute. Instead can we use `form_response` object which has errors array so we can populate multiple errors?

Reply:
As of now I would avoid using errors under form response, As we have classified 2 types of error one is submission which is a sync response to the form covering the tech error & two is the business error communicated in upcomming on_action call.